activerecord/lib/active_record/relation/predicate_builder/association_query_handler.rb

```diff
@@ -1,6 +1,17 @@
 module ActiveRecord
   class PredicateBuilder
     class AssociationQueryHandler # :nodoc:
+
+      class << self
+        def value_for(table, column, value)
+          if table.associated_with?(column).polymorphic? && ::Array === value && value.first.is_a?(Base)
```

Comment:
We shouldn't rely on the return value of `associated_with?` being anything other than truthy/falsey.